src/content_mapblock.cpp

```diff
@@ -841,19 +841,19 @@ void MapblockMeshGenerator::drawPlantlikeQuad(float rotation, float quad_offset,
 	bool offset_top_only)
 {
 	v3f vertices[4] = {
-		v3f(-scale, -BS / 2, 0),
-		v3f( scale, -BS / 2, 0),
-		v3f( scale, -BS / 2 + scale * 2, 0),
 		v3f(-scale, -BS / 2 + scale * 2, 0),
+		v3f( scale, -BS / 2 + scale * 2, 0),
+		v3f( scale, -BS / 2, 0),
+		v3f(-scale, -BS / 2, 0),
 	};
 	if (random_offset_Y) {
 		PseudoRandom yrng(face_num++ | p.X << 16 | p.Z << 8 | p.Y << 24);
 		offset.Y = BS * ((yrng.next() % 16 / 16.0) * 0.125);
 	}
-	int offset_first_index = offset_top_only ? 2 : 0;
+	int offset_first_count = offset_top_only ? 2 : 4;
```

Comment:
@paramat, that’s not that simple ;) Take a look at these snippets, which one does look better?

---
```c
	int m = offset_top_only ? 2 : 4;
	for (int i = 0; i < m; i++)
		vertices[i].Z += quad_offset;
	for (int i = 0; i < 4; i++) {
		vertices[i].rotateXZBy(rotation + rotate_degree);
		vertices[i] += offset;
	}
```
---
```c
	for (int i = 0; i < (offset_top_only ? 2 : 4); i++)
		vertices[i].Z += quad_offset;
	for (int i = 0; i < 4; i++) {
		vertices[i].rotateXZBy(rotation + rotate_degree);
		vertices[i] += offset;
	}
```
---
```c
	for (int i = 0; i < 4; i++) {
		if ((i < 2) || !offset_top_only)
			vertices[i].Z += quad_offset;
		vertices[i].rotateXZBy(rotation + rotate_degree);
		vertices[i] += offset;
	}
```
---